### PR TITLE
[ONNX] Replace optional parameters of Resize with placeholder for ops13.

### PR DESCRIPTION
**Pull request by fatcat-z**
Replace optional parameters of Resize with placeholder for ops13.

### CODE REVIEW

**Review by jiafatom — COMMENTED** [review]
Can we reuse the logic from Resize-11 exporter? The onnx spec change only affects the input to optional, code logic seems to be reusable.

**Comment by fatcat-z** [comment]
> Can we reuse the logic from Resize-11 exporter? The onnx spec change only affects the input to optional, code logic seems to be reusable.

Yes, the logic is same. Changing the dummy data used in Resize-11 exporter to place holder in Resize-13 is aimed to reduce the final ONNX file size.

**Comment by jiafatom** [comment]
> > Can we reuse the logic from Resize-11 exporter? The onnx spec change only affects the input to optional, code logic seems to be reusable.
> 
> Yes, the logic is same. Changing the dummy data used in Resize-11 exporter to place holder in Resize-13 is aimed to reduce the final ONNX file size. Place holder will cause an empty for the optional parameter while dummy data will still occupy some room iu the ONNX file.

Sorry for confusion. My point is that can we reuse the common code to avoid code duplication? Put the common code into `symbolic_helper.py`, see example as `def _squeeze_helper(g, input, axes_i):`

**Comment by fatcat-z** [comment]
> > > Can we reuse the logic from Resize-11 exporter? The onnx spec change only affects the input to optional, code logic seems to be reusable.
> > 
> > 
> > Yes, the logic is same. Changing the dummy data used in Resize-11 exporter to place holder in Resize-13 is aimed to reduce the final ONNX file size. Place holder will cause an empty for the optional parameter while dummy data will still occupy some room iu the ONNX file.
> 
> Sorry for confusion. My point is that can we reuse the common code to avoid code duplication? Put the common code into `symbolic_helper.py`, see example as `def _squeeze_helper(g, input, axes_i):`

Got it, yes it makes sense. Working on it.

File: torch/onnx/symbolic_helper.py [hunk 1]
```diff
@@ -419,6 +420,126 @@ def _interpolate_get_scales_and_mode(g, input, size, scale_factor, mode , align_
...
+                        empty_roi,
+                        scales,
+                        coordinate_transformation_mode_s=coordinate_transformation_mode,
+                        cubic_coeff_a_f=-0.75,  # only valid when mode="cubic"
+                        mode_s=interpolate_mode,  # nearest, linear, or cubic
+                        nearest_mode_s="floor")  # only valid when mode="nearest"
+    return symbolic_fn
+
+
+def __interpolate_common(g, input, size, scale_factor, mode, align_corners, recompute_scale_factor):
```
Comment (neginraoof) [inline torch/onnx/symbolic_helper.py:469]: Please rename to __interpolate_helper to keep the existing style.

File: torch/onnx/symbolic_helper.py [hunk 2]
```diff
@@ -535,6 +656,14 @@ def _is_split_static(split_size_or_sizes, _outputs):
         return False
     return True
 
+def _optional_input_placeholder_tensor(g):
+    n = g.op("prim::Constant")
+    n.setType(OptionalType.ofTensor())
+    return n
+
+def _has_more_optional_parameters():
```
Comment (neginraoof) [inline torch/onnx/symbolic_helper.py:664]: nit: This function does not have much other use-cases.
Maybe just replace this with:
if _export_onnx_opset_version >= 13

**Review by neginraoof — COMMENTED** [review]
Overall looks good. Left a couple minor comments.

File: torch/onnx/symbolic_helper.py (hunk 2 above)
Comment (fatcat-z) [reply torch/onnx/symbolic_helper.py:664]: Just feel things like '_export_onnx_opset_version >= 13' lose readability.

File: torch/onnx/symbolic_helper.py (hunk 1 above)
Comment (fatcat-z) [reply torch/onnx/symbolic_helper.py]: Sure

**Review by jiafatom — APPROVED** [review]

**Pull request merged**